internal/provider/factory_utils.go

```diff
@@ -86,6 +86,35 @@ func (f SchemaRegistryRestClientFactory) CreateSchemaRegistryRestClient(restEndp
 	}
 }
 
+type CatalogRestClientFactory struct {
+	ctx        context.Context
+	userAgent  string
+	maxRetries *int
+}
+
+func (f CatalogRestClientFactory) CreateCatalogRestClient(restEndpoint, clusterId, clusterApiKey, clusterApiSecret string, isMetadataSetInProviderBlock bool) *CatalogRestClient {
+	var opts []RetryableClientFactoryOption = []RetryableClientFactoryOption{}
+
+	// Setup DC API Client
+	dataCatalogConfig := dc.NewConfiguration()
+	if f.maxRetries != nil {
+		opts = append(opts, WithMaxRetries(*f.maxRetries))
+	}
+
+	dataCatalogConfig.UserAgent = f.userAgent
+	dataCatalogConfig.Servers[0].URL = restEndpoint
+	dataCatalogConfig.HTTPClient = NewRetryableClientFactory(f.ctx, opts...).CreateRetryableClient()
+
+	return &CatalogRestClient{
+		dataCatalogApiClient:         dc.NewAPIClient(dataCatalogConfig),
```

Comment:
nit: can we rename it to just `apiClient`: 
![image](https://github.com/user-attachments/assets/b1f3d619-a67d-4f4a-aa9a-55cfd2a5b2f4)